### PR TITLE
UI Test Input - Add "tooltip" option

### PR DESCRIPTION
**Pull request by chrblabla**
## Description

As requested in #393, a text input field now has a property called `tooltip` which can be configured. If configured, it will show a tooltip on hover in the dashboard.
For the implementation, the [tooltip component from Vuetify](https://vuetifyjs.com/en/components/tooltips) has been used.

## Related Issue(s)

#393

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by chrblabla** [comment]
Example:

Bildschirmfoto 2024-07-04 um 18 32 01

**Comment by joepavitt** [comment]
Thanks for the PR @chrblabla - this works great with a "tooltip" defined, but breaks if one is not... I'll try and put in a quick fix so we can get it merged

**Comment by chrblabla** [comment]
Hi @joepavitt, thank you for your comment. Do you mind elaborating what exactly breaks?

**Comment by joepavitt** [comment]
For migrating from older versions, there will be existing `ui-text-input` nodes that have `props.tooltip` as `undefined`, so the `tooltip.length` on your `:disabled` logic falls over.

I'm putting in a fix as we speak

File: ui/src/widgets/ui-text-input/UITextInput.vue [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-        :disabled="!state.enabled" class="nrdb-ui-text-field"
-        :label="label" :type="type" :rules="validation" variant="outlined" hide-details="auto" @update:model-value="onChange" @keyup.enter="onEnter" @blur="onBlur"
-    />
-    <v-textarea
-        v-else
-        v-model="value" :disabled="!state.enabled" class="nrdb-ui-text-field"
-        :label="label" variant="outlined" hide-details="auto" @update:model-value="onChange" @blur="send"
-    />
+    <v-tooltip :text="tooltip" :disabled="tooltip.length === 0" location="bottom">
+        <template v-slot:activator="{ props }">
```
Comment (joepavitt) [inline ui/src/widgets/ui-text-input/UITextInput.vue:3]:
```suggestion
    <v-tooltip :text="tooltip" :disabled="!tooltip?.length" location="bottom">
        <!-- eslint-disable-next-line vue/no-template-shadow -->
        <template v-slot:activator="{ props }">
```

Safer logic check for older `ui-text-input` nodes that would have no `tooltip` defined. Also added a `eslint-disable` as it's complaining about the use of `props` here (which is the correct usage)

Comment (joepavitt) [reply ui/src/widgets/ui-text-input/UITextInput.vue:3]: @chrblabla if you can merge/accept this suggestion, then it's good to go. I don't have permissions to push to your branch directly, so couldn't resolve it myself

**Comment by chrblabla** [comment]
@joepavitt thank you for the explanation and good catch 🙂 I accepted the suggestion and updated the branch to also reflect the recently merged icon feature.

**Review by joepavitt — APPROVED** [review]

**Pull request merged**